client/notifier/channel-notifier.go

```diff
@@ -0,0 +1,90 @@
+package notifier
+
+import (
+	"github.com/statechannels/go-nitro/client/engine/store"
+	"github.com/statechannels/go-nitro/client/query"
+	"github.com/statechannels/go-nitro/internal/safesync"
+	"github.com/statechannels/go-nitro/payments"
+	"github.com/statechannels/go-nitro/types"
+)
+
+const ALL_NOTIFICATIONS = "all"
+
+// ChannelNotifier is used to notify multiple listeners of a channel update.
+type ChannelNotifier struct {
+	ledgerListeners  *safesync.Map[*ledgerChannelListeners]
+	paymentListeners *safesync.Map[*paymentChannelListeners]
+	store            store.Store
+	vm               *payments.VoucherManager
+}
+
+// NewChannelNotifier constructs a channel notifier using the provided store.
+func NewChannelNotifier(store store.Store, vm *payments.VoucherManager) *ChannelNotifier {
+	return &ChannelNotifier{
+		ledgerListeners:  &safesync.Map[*ledgerChannelListeners]{},
+		paymentListeners: &safesync.Map[*paymentChannelListeners]{},
+		store:            store,
+		vm:               vm,
+	}
+}
+
+// RegisterForAllLedgerUpdates returns a buffered channel that will receive ledger channel updates when they occur for all ledger channels.
```

Comment:
```suggestion
// RegisterForAllLedgerUpdates returns a buffered channel that will receive updates for all ledger channels.
```
Slight improvement removing ambiguity.